File: 2024/06/p1.test.lua
local describe = require("lib.test")
local p1 = require(".2024.06.p1")

describe("input", function(it)
  it("example", function()
  local example = [[....#.....
.........#
..........
..#.......
.......#..
..........
.#..^.....
........#.
#.........
......#...]]
    local expected = 41

    local n = p1.solve(example)

    assert(n == expected, string.format("Expected %d, but was %d", expected, n))
  end)

  it("input", function()
    local fp = io.open("./2024/06/assets/input.txt", "r") or error("Input not found")
    local input = fp:read("*a")
    local expected = 5162

    local n = p1.solve(input)

    assert(n == expected, string.format("Expected %d, but was %d", expected, n))
  end)
end)

File: 2024/12/__tests__/p1.test.ts
import assert from "node:assert";
import { readFile } from "node:fs/promises";
import { describe, it } from "node:test";
import { solve } from "../p1.js";
import path from "node:path";

describe("p1", () => {
  it("example 1", () => {
const example = `AAAA
BBCD
BBCC
EEEC`;

    assert.strictEqual(solve(example), 140);
  });

  it("example 2", () => {
const example = `OOOOO
OXOXO
OOOOO
OXOXO
OOOOO`;

    assert.strictEqual(solve(example), 772);
  });

  it("example 3", () => {
const example = `RRRRIICCFF
RRRRIICCCF
VVRRRCCFFF
VVRCCCJFFF
VVVVCJJCFE
VVIVCCJJEE
VVIIICJJEE
MIIIIIJJEE
MIIISIJEEE
MMMISSJEEE`;

    assert.strictEqual(solve(example), 1930);
  });

  it("input", async () => {
    const filename = path.join(import.meta.dirname, "../assets/input.txt").replace("/dist/", "/");
    const input = await readFile(filename, "utf8");

    assert.strictEqual(solve(input), 1457298);
  });
});
